addons/website_sale/static/src/website_builder/dynamic_snippet_category_option_plugin.js

```diff
@@ -0,0 +1,64 @@
+import { Plugin } from '@html_editor/plugin';
+import { withSequence } from '@html_editor/utils/resource';
+import { BuilderAction } from "@html_builder/core/builder_action";
+import { registry } from '@web/core/registry';
+import { DEVICE_VISIBILITY } from '@website/builder/option_sequence';
+import { setDatasetIfUndefined } from '@website/builder/plugins/options/dynamic_snippet_option_plugin';
+import { DynamicSnippetCategoryOption } from './dynamic_snippet_category_options';
+
+const TEMPLATE_KEY = {
+    'clickable': 'website_sale.dynamic_filter_template_product_public_category_clickable_items',
+    'default': 'website_sale.dynamic_filter_template_product_public_category_default',
+}
+
+export class DynamicSnippetCategoryOptionPlugin extends Plugin {
+    static id = 'dynamicSnippetCategoryOptionPlugin';
+    static dependencies = ['dynamicSnippetOption'];
+    selector = 'section.s_dynamic_snippet_category'
+    modelNameFilter = 'product.public.category';
+    resources = {
+        builder_options: [
+            withSequence(DEVICE_VISIBILITY, {
+                OptionComponent: DynamicSnippetCategoryOption,
+                props: {
+                    modelNameFilter: this.modelNameFilter,
+                },
+                selector: this.selector,
+                groups: ['website.group_website_designer'],
+            }),
+        ],
+        builder_actions: { toggleClickableAction },
+        on_snippet_dropped_handlers: this.onSnippetDropped.bind(this),
+    };
+
+    async onSnippetDropped({ snippetEl }) {
+        for (const [optionName, value] of [
+            ['showParent', 'true'],
+            ['columns', 2],
+            ['size', 'medium'],
+            ['button', 'Explore Now'],
+            ['alignment', 'left'],
+        ]) {
+            setDatasetIfUndefined(snippetEl, optionName, value);
+        }
+        await this.dependencies.dynamicSnippetOption.setOptionsDefaultValues(
+            snippetEl,
+            this.modelNameFilter,
+            [],
+        );
+    }
+}
+
+export class toggleClickableAction extends BuilderAction {
+    static id = 'toggleClickable';
+    apply({ editingElement, value }) {
+        const nodeData = editingElement.dataset;
+        nodeData.templateKey = nodeData.templateKey == TEMPLATE_KEY['default']
+            ? 'website_sale.dynamic_filter_template_product_public_category_clickable_items'
+            : 'website_sale.dynamic_filter_template_product_public_category_default';
```

Comment:
why can not we get the value of the checkbox?
```suggestion
        nodeData.templateKey = nodeData.templateKey == TEMPLATE_KEY['default']
            ? TEMPLATE_KEY['clickable']
            : TEMPLATE_KEY['default'];
```